### DISCUSSION TITLE
Media Upload

### DISCUSSION THREAD

**Discussion by jemclean** [noweh/twitter-api-v2-php#13] [Q&A] [answered]
I am new to the twitter api. Do you have example code on how to upload a photo (png) when posting to a timeline?

**Comment by noweh** [comment, marked as answer]
Hello and welcome !
as specified in this issue [here](https://github.com/noweh/twitter-api-v2-php/issues/3), it is still not possible to upload media in version 2 of the Twitter API.

There is a trick (not perfect): use the url of an image that already exists on the internet in the text of the message.

Regards

**Reply by jemclean** [reply]
Thanks for the information. I just tried the following and I am only seeing the url text being posted. Any particular way the url needs to be formatted?

$return = $client->tweet()->performRequest('POST', ['text' => "https://source.unsplash.com/random/200x200?sig=1"]);

**Reply by noweh** [reply]
Indeed, it does nothing with this url. I use this for this kind of post and it works : https://twitter.com/SteveBOTgers/status/1548021579749068803?t=6X1SbLyeTXSolWI0x6eRAA&s=01
